e2e_tests/tests/experiment/test_core.py

```diff
@@ -52,17 +52,14 @@ def test_metric_gathering() -> None:
     # Check training metrics.
     full_trial_metrics = exp.trial_metrics(trials[0].trial.id)
     batches_trained = 0
-    for step in full_trial_metrics["steps"]:
-        metrics = step["metrics"]
-
-        actual = metrics["batch_metrics"]
-        assert len(actual) == scheduling_unit
+    for step in full_trial_metrics:
+        assert step.batches == batches_trained + scheduling_unit
```

Comment:
this change doesn't make sense to me. the new check is very different, and doesn't test the same as the old.